ocpp1.6/charge_point.go

```diff
@@ -227,11 +227,15 @@ func (cp *chargePoint) SendRequest(request ocpp.Request) (ocpp.Response, error)
 	if err != nil {
 		return nil, err
 	}
-	asyncResult, ok := <-asyncResponseC
-	if !ok {
-		return nil, fmt.Errorf("internal error while receiving result for %v request", request.GetFeatureName())
+	select {
+	case asyncResult, ok := <-asyncResponseC:
+		if !ok {
+			return nil, fmt.Errorf("internal error while receiving result for %v request", request.GetFeatureName())
+		}
+		return asyncResult.r, asyncResult.e
+	case <-cp.stopC:
```

Comment:
Sadly doesn't guarantee to work, since there's already another goroutine waiting for the `stopC` signal (see `asyncCallbackHandler`).

The issue is actually the `invokeCallback` flag, passed to:
```
func (cp *chargePoint) clearCallbacks
``` 
When a client is stopped, the internal callback queue would be emptied (ideally also unblocking anyone sitting in the `SendRequest` call). I decided to not invoke every pending callback when a user explicitly invokes `Stop`, since it "generates spam" on the user's side. 
I realize now, that there is that edge case, and all pending callbacks should probably be invoked nevertheless on the client. WDYT?

Reply:
> Sadly doesn't guarantee to work, since there's already another goroutine waiting for the stopC signal (see asyncCallbackHandler).

Should be ok though since we `close(stopC)`, it will receive everywhere

Reply:
Right, fair enough. I'm merging this one.